### PR TITLE
User can delete their  past post

### PR DESCRIPTION
**Pull request by rithreaksa**
[User can delete their  past post](https://www.pivotaltracker.com/story/show/176626432)

### User Story
```
As a user,            
In order to keep my feed in a good state,         
I want to delete my posts.
```
## Changes proposed in this pull request:
* Creates a test to check if user can delete their past post
* Implements delete function 

## What I have learned working on this feature:
* How to delete a post from post index
* How to implement filter action

## Screenshots (Client)

Screen Shot 2021-02-05 at 10 22 15 AM

### CODE REVIEW

File: src/modules/PostService.js [hunk 1]
```diff
@@ -28,6 +28,24 @@ const PostService = {
       })
     }
   },
+  async delete(postId, credentials) {
+    try {
+      let response = await axios.delete(
+        `${API_URL}/posts/${postId}`,
+        { post_id: postId },
+        { headers: credentials }
```
Comment (sealfury) [inline src/modules/PostService.js]: we're not accepting credentials in the backend, so we can probably remove it here

File: src/state/reducers/rootReducer.js [hunk 2]
```diff
@@ -40,13 +40,18 @@ const rootReducer = (state, action) => {
...
     case 'SET_USER_ID':
       return {
         ...state,
         userId: action.payload,
       }
+    case 'SET_DELETE_POST':
+      return {
+        ...state,
+        userPosts: state.userPosts.filter(item => item.id !== action.payload),
+      }
```
Comment (sealfury) [inline src/state/reducers/rootReducer.js:54]: nice solution here!

File: cypress/integration/userCanDeleteTheirPost.feature.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    cy.get('[data-testid=login-screen]').within(() => {
+      cy.get('[data-testid=login-email]').type('spotifyuser@spotify.com')
+      cy.get('[data-testid=login-password]').type('password')
+      cy.get('[data-testid=login-submit]').click()
+    })
+    cy.contains('Profile').click()
+  })
+
+  it('successfully delete their post', () => {
+    cy.get('[data-testid=user-post-1]').should('contain', 'Rosa helikopter')
```
Comment (KyunginNa) [inline cypress/integration/userCanDeleteTheirPost.feature.js:42]: I think you can delete this assertion! We are checking it in the post index test. :)

File: src/components/UserProfile.jsx [hunk 4]
```diff
@@ -35,6 +36,10 @@ const UserProfile = () => {
               <ListItem.Title>{item.track}</ListItem.Title>
               <ListItem.Subtitle>{item.artists}</ListItem.Subtitle>
             </ListItem.Content>
+            <Button 
+              testID={`delete-button-${item.id}`}
+              onPress={() => {PostService.delete(item.id, credentials)}}
+            >Delete</Button>
```
Comment (sealfury) [inline src/components/UserProfile.jsx:42]: just a note to say we can fix this in styling later

File: cypress/integration/userCanDeleteTheirPost.feature.js (hunk 3 above)
Comment (rithreaksa) [reply cypress/integration/userCanDeleteTheirPost.feature.js:42]: yes ma'am!!

File: src/components/UserProfile.jsx (hunk 4 above)
Comment (rithreaksa) [reply src/components/UserProfile.jsx:42]: yep :)

File: src/modules/PostService.js (hunk 1 above)
Comment (rithreaksa) [reply src/modules/PostService.js]: Yes! Got it! Thanks.

File: src/state/reducers/rootReducer.js (hunk 2 above)
Comment (rithreaksa) [reply src/state/reducers/rootReducer.js:54]: google!!!! :D

**Pull request merged**